### ISSUE TITLE
Zero values as note_id in cache_notes table

### ISSUE THREAD

**Issue by deg-pl** [opencaching/opencaching-pl#2132]
Most of cache user notes has 0 as note_id field.

**Comment by kojoty** [comment]
note_id is not needed - I'm not sure but I guess that only OKAPI sets this (?)

The primary key is a <cache_id><user-id> pair and this is OK - we don't need separate key for this table as long as we have a assumption that the user can add only one note to the cache and the new note erased the previous version of the not (what in my opinion is OK.

Let it open for deeper investigation.

**Comment by deg-pl** [comment]
I think
- if we really don't need it - let's remove it
- if we need it -let's fix it ;)

**Comment by rapotek** [comment]
The `note_id` column in `cache_notes` table is used in OKAPI only (`services/caches/save_personal_notes/WebService.php`, lines 152-191) but in a strange way. It looks like it is a leftover from a previous table definition. If there are no duplicates for [`cache_id`, `user_id`] pair in the table (there shouldn't be any since the pair is a primary key), this okapi code can be safely modified to exclude the `note_id` column completely an then the column can be dropped.

**Comment by kojoty** [comment]
yes, but please note that OKAPI is also used by OCDE and I'm not sure about
the structure of their DB - but this is right generally.

**Comment by rapotek** [comment]
There are separate "ifs" for OCDE and OCPL in the code, so it can be safely modified.
